src/cli/argument-parser/index.ts

```diff
@@ -454,9 +455,12 @@ export default class CLIArgumentParser {
             el => optionNames.some(opt => el.startsWith(opt)));
 
         if (optionIndex > -1) {
-            const isNotLastOption       = optionIndex < argv.length - 1;
-            const shouldMoveOptionToEnd = isNotLastOption &&
-                !subOptionsNames.some(opt => argv[optionIndex + 1].startsWith(opt));
+            const isNotLastOption        = optionIndex < argv.length - 1;
```

Comment:
Move this code to the separate utility function `cli/util/should-move-option-to-end.js` and add tests to it.